packages/modules/src/lumyComponent.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { registerLumyComponent } from '@dharpa-vre/client-core'
+import { createGenerateClassName, StylesProvider } from '@material-ui/styles'
+import { ThemeContextProvider } from '@lumy/styles'
+
+/**
+ * Register Lumy component under a specified ID.
+ * Namespace CSS classes under a uniq prefix created from the ID.
+ */
+export function lumyComponent(id: string) {
+  return <T,>(Component: React.FC<T>): React.FC<T> => {
+    const generateClassName = createGenerateClassName({
+      productionPrefix: `${id}-`,
+      disableGlobal: false,
+      seed: `${id}-`
+    })
+
+    const enrichedComponent = (props: T) => (
```

Comment:
Creative solution!

Reply:
It's the only one implicitly recommended by Material UI documentation :) 
And, you reminded me that I forgot to move it to a separate package, which I just did.